src/include/mpi.h.in

```diff
@@ -581,14 +581,14 @@ static const MPI_Datatype mpich_mpi_offset MPICH_ATTR_TYPE_TAG(MPI_Offset) = MPI
 /* The order of these elements must match that in mpif.h, mpi_f08_types.f90,
    and mpi_c_interface_types.f90 */
 typedef struct MPI_Status {
-    int count_lo;
-    int count_hi_and_cancelled;
-    int MPI_SOURCE;
-    int MPI_TAG;
-    int MPI_ERROR;
+    MPI_Fint count_lo;
```

Comment:
This is really weird to me. If we are insisting that a Fortran INTEGER is the same as a C int, why use this typedef?

Reply:
We are not insisting that MPI_Fint is the same as a C int. We are just not sure that whether it will work correctly when MPI_Fint is not the same size as C int since that is hardly ever (let alone thoroughly) tested. It just appears to me that if we directly use `MPI_Fint` in `MPI_Status`, then at least for the status object its compatibility would be guaranteed and thus much safer. So above change is reasoned from the *opposite* of insisting that `MPI_Fint` is the same as `int`.

I see your concern of `ABI` change. I guess it is not worse than what current code assumes. Let me abandon this change and opt for the simplest for now.